modules/ampir/main.nf

```diff
@@ -0,0 +1,47 @@
+process AMPIR {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::r-ampir=1.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/r-ampir:1.1.0':
+        'quay.io/biocontainers/r-ampir:1.1.0' }"
+
+    input:
+    tuple val(meta), path(faa)
+    val model
+    val min_length
+    val min_probability
+
+    output:
+    tuple val(meta), path("*.faa"), emit: amps_faa
+    tuple val(meta), path("*.csv"), emit: amps_csv
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    min_length = ("${min_length}" == "[]") ? "": " min_len = as.integer(${min_length})," // Fall back to AMPir default value if none specified
+    """
+    #!/usr/bin/env Rscript
+    library(ampir)
+
+    input_seqs <- read_faa('${faa}')
+    prediction <- predict_amps(input_seqs,${min_length} model = '${model}')
+    prediction <- prediction[which(prediction\$prob_AMP >= as.numeric(${min_probability})), ]
+    output_seqs <- input_seqs[row.names(prediction), ]
+    write.table(prediction, file = "${prefix}.csv", row.names = FALSE, sep = ";", quote = FALSE, dec = '.')
```

Comment:
```suggestion
    write.table(prediction, file = "${prefix}.csv", row.names = FALSE, sep = ",", quote = FALSE, dec = '.')
```

Shouldn't it be `,` if CSV? If there is a conflict wiht the contents maybe try with tab instead?

Reply:
Of course, FASTA headers will always contain any character you can imagine 🥴 Tab might be the safest separator.